egg-herbie/main.rkt

```diff
@@ -70,30 +156,22 @@
                (register-finalizer p egraph_destroy)
                p)))
 
-; Egraph iteration data
-; Not managed by Racket GC.
-; Must call `destroy_egraphiters` to free.
-(define-cstruct _EGraphIter ([numnodes _uint] [numeclasses _uint] [time _double]) #:malloc-mode 'raw)
-
-; Rewrite rule
-; Not managed by Racket GC.
-; Must call `free` on struct and fields
-(define-cstruct _FFIRule ([name _pointer] [left _pointer] [right _pointer]) #:malloc-mode 'raw)
-
-;;  -> a pointer to an egraph
+;; Constructs an e-graph instances.
 (define-eggmath egraph_create (_fun -> _egraph-pointer))
 
+;; Frees an e-graph instance.
 (define-eggmath egraph_destroy (_fun _egraph-pointer -> _void))
 
-(define-eggmath destroy_string (_fun _pointer -> _void))
-
-;; egraph pointer, s-expr string -> node number
-(define-eggmath egraph_add_expr (_fun _egraph-pointer _string/utf-8 -> _uint))
+;; Copies an e-graph instance.
+(define-eggmath egraph_copy (_fun _egraph-pointer -> _egraph-pointer))
```

Comment:
Do we use this anywhere?

Reply:
Running a rule schedule requires copying the e-graph to reset all the iteration data.

Reply:
Oh. Ugly but ok